homeassistant/components/tahoma/cover.py

```diff
@@ -0,0 +1,359 @@
+            CORE_TARGET_CLOSURE_STATE,
+            CORE_CLOSURE_OR_ROCKER_POSITION_STATE,
+        )
+
+        # Uno devices can have a position not in 0 to 100 range when unknown
+        if position is None or position < 0 or position > 100:
+            return None
+
+        if not self._reversed_position_device():
+            position = 100 - position
+
+        return position
+
+    @property
+    def current_cover_tilt_position(self):
+        """Return current position of cover tilt.
+
+        None is unknown, 0 is closed, 100 is fully open.
+        """
+        position = self.executor.select_state(
+            CORE_SLATS_ORIENTATION_STATE, CORE_SLATE_ORIENTATION_STATE
+        )
+        return 100 - position if position is not None else None
+
+    async def async_set_cover_position(self, **kwargs):
+        """Move the cover to a specific position."""
+        position = 100 - kwargs.get(ATTR_POSITION, 0)
+
+        if self._reversed_position_device():
+            position = kwargs.get(ATTR_POSITION, 0)
+
+        await self.executor.async_execute_command(
+            self.executor.select_command(*COMMANDS_SET_POSITION), position
+        )
+
+    async def async_set_cover_tilt_position(self, **kwargs):
+        """Move the cover tilt to a specific position."""
+        await self.executor.async_execute_command(
+            self.executor.select_command(*COMMANDS_SET_TILT_POSITION),
+            100 - kwargs.get(ATTR_TILT_POSITION, 0),
+        )
+
+    @property
+    def is_closed(self):
+        """Return if the cover is closed."""
+        state = self.executor.select_state(
+            CORE_OPEN_CLOSED_STATE,
+            CORE_SLATS_OPEN_CLOSED_STATE,
+            CORE_OPEN_CLOSED_PARTIAL_STATE,
+            CORE_OPEN_CLOSED_PEDESTRIAN_STATE,
+            CORE_OPEN_CLOSED_UNKNOWN_STATE,
+            MYFOX_SHUTTER_STATUS_STATE,
+        )
+
+        if state is not None:
+            return state == STATE_CLOSED
+
+        if self.current_cover_position is not None:
+            return self.current_cover_position == 0
+
+        if self.current_cover_tilt_position is not None:
+            return self.current_cover_tilt_position == 0
+
+        return None
+
+    @property
+    def device_class(self):
+        """Return the class of the device."""
+        return (
+            TAHOMA_COVER_DEVICE_CLASSES.get(self.device.widget)
+            or TAHOMA_COVER_DEVICE_CLASSES.get(self.device.ui_class)
+            or DEVICE_CLASS_BLIND
+        )
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend, if any."""
+        if (
+            self.executor.has_state(CORE_PRIORITY_LOCK_TIMER_STATE)
+            and self.executor.select_state(CORE_PRIORITY_LOCK_TIMER_STATE) > 0
+        ):
+            if self.executor.select_state(IO_PRIORITY_LOCK_ORIGINATOR_STATE) == "wind":
+                return ICON_WEATHER_WINDY
+            return ICON_LOCK_ALERT
```

Comment:
Based on previous feedback, I understood that changing the `icon` is not allowed. What would be the best way to show that the device is in a 'locked' state, thus that commands won't be executed?

Reply:
Do we know the state of the cover when it's locked, ie open/closed?

Reply:
Yes. Only `io` devices (two way communication) can be locked due to an external reason (wind, snow etc.) and those devices always have their state available. 

It should be possible to override some of the locked positions via the official vendor app, but we haven't implemented this for HA. It would be good to indicate that all commands you execute via HA, will not succeed. For now, we use an icon since this was used in the current `tahoma` integration in core as well.

Reply:
My suggestion is that we raise an error in the service handler methods when a device is locked and the command won't succeed. This will show in the GUI to the user. And we drop the custom icon.

Reply:
I have removed this functionality for now. Let's see if users really use this and otherwise add it in a follow up PR.

The obstruction detected attribute will be added for those cases already, thus it is visible.